addons/purchase_requisition/wizard/purchase_requisition_create_alternative.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models, Command
+from odoo.exceptions import UserError
+
+
+class PurchaseRequisitionCreateAlternative(models.TransientModel):
+    _name = 'purchase.requisition.create.alternative'
+    _description = 'Wizard to preset values for alternative PO'
+
+    origin_po_id = fields.Many2one(
+        'purchase.order', help="The original PO that this alternative PO is being created for."
+    )
+    partner_id = fields.Many2one(
+        'res.partner', string='Vendor', required=True,
+        help="Choose a vendor for alternative PO")
+    creation_blocked = fields.Boolean(
+        help="If the chosen vendor or if any of the products in the original PO have a blocking warning then we prevent creation of alternative PO. "
+             "This is because normally these fields are cleared w/warning message within form view, but we cannot recreate that in this case.",
+        compute="_compute_purchase_warn",
+        groups="purchase.group_warning_purchase")
+    purchase_warn_msg = fields.Text(
+        'Warning Messages',
+        compute="_compute_purchase_warn",
+        groups="purchase.group_warning_purchase")
+    copy_products = fields.Boolean(
+        "Copy Products", default=True,
+        help="If this is checked, the product quantites of the original PO will be copied")
+
+    @api.depends('partner_id', 'copy_products')
+    def _compute_purchase_warn(self):
+        self.creation_blocked = False
+        self.purchase_warn_msg = ''
+        # follows partner warning logic from PurchaseOrder
+        if self.env.user.has_group('purchase.group_warning_purchase'):
+            if self.partner_id and self.partner_id.purchase_warn != 'no-message':
+                partner = self.partner_id
+                if self.partner_id.purchase_warn == 'no-message':
+                    partner = partner.parent_id
```

Comment:
This condition will always be false since we check `purchase_warn` is different than `'no-message'` two lines upper:
```python3
if self.partner_id and self.partner_id.purchase_warn != 'no-message':
```